rust/foxglove/src/mcap_writer.rs

```diff
@@ -8,22 +8,32 @@ use std::{fmt::Debug, io::Write};
 
 use crate::library_version::get_library_version;
 use crate::sink_channel_filter::SinkChannelFilterFn;
-use crate::{ChannelDescriptor, Context, FoxgloveError, Sink, SinkChannelFilter};
+use crate::{ChannelDescriptor, Context, FoxgloveError, Sink, SinkChannelFilter, SinkId};
 
 /// Compression options for content in an MCAP file
 pub use mcap::Compression as McapCompression;
 /// Options for use with an [`McapWriter`][crate::McapWriter].
 pub use mcap::WriteOptions as McapWriteOptions;
 
+#[cfg(feature = "live_visualization")]
```

Comment:
Your changes don't seem related to the live_visualization feature flag, I'd remove this attribute